gateway/ln-gateway/src/client.rs

```diff
@@ -15,6 +15,40 @@ use tracing::{debug, warn};
 
 use crate::{LnGatewayError, Result};
 
+pub trait IDbFactory: Debug {
+    fn create_database(&self, federation_id: FederationId, _path: PathBuf) -> Result<Database>;
```

Comment:
Its primary function is to open the DB not create it, right? (creating might be a prerequisite)

Reply:
Ah yes. Though I'm not sure which one we should highlight, with the naming. The actual behavior to strictly depend on the DB type and implementation. For example,  right now `MemDbFactory` impl will always create a new instance, `RocksDbFactory` or `SledDbFactory` might simply open an existing db, or create a new instance if none is present